.github/workflows/master-check.yml

```diff
@@ -0,0 +1,56 @@
+name: Check commits to master
+
+on:
+    pull_request:
+        branches:
+            - master
```

Comment:
This eventually will be triggered on commits to `master` branch, two things I guess we could improve:

*   Would it be possible to detect `[skip ci]` in the commit message which worked with Travis and we are used to using it (e.g. when updating changelog). So we can skip test runs for some commits.
*   Apart from master we could add `stable` branch and `vA.B.C` tags too.

WDYT?

Reply:
Yeah, that makes sense. It's already supported by GH actions -> see [here](https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/). I was thinking about adding `paths` filter so only changes to certain directories would trigger pipline but that requires additional maintenance of a whitelist.

So my suggestion is to try to leverage [this feature](https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/) and see if that's enough.

In addition, I will add triggers for `stable` branch and version tags.

Reply:
BTW, `paths` are described [here](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths).

Reply:
We can go with `[skip ci]` for now, I see it works out-of-the-box :+1:

The `paths` thing is an interesting one. I think we could benefit from using it too (not having to remember about `skip ci` is one think less to remember about). @MMMalik could you extract an issue for that and maybe we can have it in v2? It will be easier to use if we will be having it in mind from the very beginning when working on test infrastructure. WDYT?

Reply:
Added task #197